src/audio/pipewire.c

```diff
@@ -324,6 +324,13 @@ static int pipewire_feed_sink_overlap(AudioID *id, AudioTrack track)
         samples_buffer_copy += room_left_in_buffer;
         message(4, "writing completed without errors");
     }
+    // because this is feed_sink_overlap, we wait till the ringbuffer falls below 10 ms, to not lose some small segments of speech
+    fill_quantity = 0;                                                     // reset
+    while (fill_quantity >= track.sample_rate * state->stride * 10 / 1000) // untill there's less than 10 ms in the ringbuffer
+    {
+        fill_quantity = spa_ringbuffer_get_write_index(&state->rb, &write_index); // update the value
```

Comment:
you want to update the value *after* calling spa_system_eventfd_read, not before.

And as it is now (fill_quantity initialized to 0), the loop will not actually do anything: fill_quantity being 0, it cannot be larger than 10ms of audio data.

Reply:
oops, weird that I still felt some kind of latency from it though, maybe this is software bug placebo, lol

Reply:
fixed